File: design-patterns/java/strategy/ThirdPartyApiShipping.java
public class ThirdPartyApiShipping implements ShippingStrategy {
    private final double baseFee;
    private final double percentageFee;

    public ThirdPartyApiShipping(double baseFee, double percentageFee) {
        this.baseFee = baseFee;
        this.percentageFee = percentageFee;
    }

    @Override
    public double calculateCost(Order order) {
        System.out.println("Calculating with Third-Party API strategy.");
        // Simulate API call
        return baseFee + (order.getOrderValue() * percentageFee);
    }
}

File: design-patterns/cpp/strategy/third_party_api_shipping.cpp
#include "third_party_api_shipping.h"
#include <iostream>

double ThirdPartyApiShipping::calculateCost(const Order& order) {
    std::cout << "Calculating shipping cost using third-party API..." << std::endl;
    // Simulate API call
    return order.getOrderValue() * 0.1; // 10% of order value
} 